nni/retiarii/experiment/pytorch.py

```diff
@@ -106,15 +107,20 @@ def _validation_rules(self):
     'training_service': lambda value: (type(value) is not TrainingServiceConfig, 'cannot be abstract base class')
 }
 
-def preprocess_model(base_model, trainer, applied_mutators, full_ir=True):
+def preprocess_model(base_model, trainer, applied_mutators, full_ir=True, 
+                     parse_shape=False, example_inputs=None):
     # TODO: this logic might need to be refactored into execution engine
     if full_ir:
         try:
             script_module = torch.jit.script(base_model)
         except Exception as e:
             _logger.error('Your base model cannot be parsed by torch.jit.script, please fix the following error:')
             raise e
-        base_model_ir = convert_to_graph(script_module, base_model)
+        if parse_shape:
```

Comment:
This looks like another execution engine.
I suppose you can merge `parse_shape` and `example_inputs` with `full_ir`, and rename `full_ir` to something like `ir_format`.